design/src/rfcs/rfc0022_error_context_and_compatibility.md

```diff
@@ -0,0 +1,370 @@
+RFC: Error Context and Compatibility
+====================================
+
+> Status: RFC
+>
+> Applies to: Generated clients and shared rust-runtime crates
+
+This RFC proposes a pattern for writing Rust errors to provide consistent
+error context AND forwards/backwards compatibility. The goal is to strike
+a balance between these four goals:
+
+1. Errors are forwards compatible, and changes to errors are backwards compatible
+2. Errors are ergonomic
```

Comment:
I think it would be helpful to be more specific here somehow. What does it mean for an error to be ergonomic? Does it have helpful methods? Is it easy to match on? Is it easy to figure out how I can use it to understand the problem that necessitated it?